In [1]:
import pandas as pd
import numpy as np

# S11_L58

You are given the data from the lesson
Test the null hypothesis at 10% significance

We have known variance of the population which is 225000000 and with STD it is 15000

The H0 is the data scientist salary is 113 000 USD on average, and the alternative is that it differs from 113 000 USD

In [2]:
salaries = pd.read_excel("S11_L58\salary.xlsx")
salaries.head()

Dataset
0   117313
1   104002
2   113038
3   101936
4    84560

**Mean**

In [3]:
round(salaries.mean())

Dataset    100200.0
dtype: float64

In [4]:
mean_sample =  round(salaries.mean())

**Standard error**

In [5]:
round(15000 / np.sqrt(len(salaries))) # remember we are considering the STD of the population

2739

In [6]:
SE = round(15000 / np.sqrt(len(salaries)))

**Sample size**

In [7]:
len(salaries)

30

10% level of significance it is alpha 0.10 / 2 which is 0.05 since it is atwo sided test issue and 1 - 0.05 is 0.95, this 0.95 is capital Z, and 1.65 is the sum of the axes on the Z table, this is lower z this 1.65

**Time for calculating te Z test**

The hypothesized mean is 113 000 USD as the average salary and 100 200 USD is the mean of the sample

In [8]:
Z_test = round((mean_sample - 113000) / SE, 2)

In [9]:
Z_test

Dataset   -4.67
dtype: float64

1.65 < 4.67 which means we can reject the H0 with a confidence interval of 0.10, in the refered lesson with a significance level alpha of 0.5 the critical value was 1.96, the Z test is still higher than the cut of point 1.96 < 4.67. There is no statistical evidence to accept H0 with the provided data. 

# S11_L61

Background	You are given the data from the lesson.							

1. Task 1	What if the question was: is the competitor open rate EXACTLY 40%. What would be the decision then?							
2. Task 1.1	Test at 5% significance. Comment on the decision with the appropriate statistical jargon.						
3. Task 1.2	Test at 1% significance. Comment on the decision with the appropriate statistical jargon.						
4. Task 1.3	Calculate the p-value of the test. Solve 1.1 and 1.2 using the p-value						

In [10]:
open_rate = pd.read_excel("S11_L61\open_rate.xlsx")
open_rate.head()

Open rate
0       0.26
1       0.23
2       0.42
3       0.49
4       0.23

# Competitor open rate = 40% is a two sided tail hypothesis 

We have an unknown population variance and we are going to use the T statistics distribution and table

**Mean**

In [11]:
open_rate.mean()

Open rate    0.377
dtype: float64

In [12]:
mean2 = open_rate.mean()
mean2

Open rate    0.377
dtype: float64

**Standard error**

In [13]:
round(open_rate.std() / np.sqrt(len(open_rate)), 3)

Open rate    0.043
dtype: float64

In [14]:
SE2 = round(open_rate.std() / np.sqrt(len(open_rate)), 3)
SE2

Open rate    0.043
dtype: float64

**lenght of the data**

In [15]:
len_op = len(open_rate)
len_op

10

**Calculating this time the T-score**

In [16]:
T_score = round((mean2 - 0.4) / SE2, 2)
T_score

Open rate   -0.53
dtype: float64

We are going to test at 5% level of significance, alpha is = to 0.05 but since it is a two tailed H0 we divide it by two which would be equal to 0.025. As the data has lenght 10 - 9 it would be 9, hence the T table critical value would be 2.26.

The absolute value of the T score is less than the critical value, T socre (0.53) < T critical (2.26), for this reason we accept H0

**We are going to test now 1% with significance level on a two sided H0**

0.01 / 2  = 0.005, the critical value of the T table is 3.25 and the T core is still lower than 3.25, the significance level of 1%, we still accept the H0.

The p-value of this test is 0.608 > 0.05 and 0.01, therefore, we cannot reject the null hypothesis. In a two sided test, and 0.304 in a one sided test, in both cases the value is greater than alpha, we accept H0. 

# S11_L63

Background	Some guy on the internet designed a weight-loss program. You are wondering if it is working. You are given a sample of some people who did the program. You can find the data in kg if you prefer working with kg as a unit of measurement.												
1. Task 1	Calculate the difference between before and after												
2. Task 2	State the null hypothesis.												
3. Task 3	Calculate the appropriate statistic												
4. Task 4	Decide if this is a one-sided or a two-sided test. Calculate the p-value of this statistic							

5. Task 5	Based on the p-value, decide at 1%,5% and 10% significance, if the program is working. Comment using the appropriate statistical jargon.												


In [17]:
weights = pd.read_excel("S11_L63\weights.xlsx")
weights.head()

Before (lbs)  After (lbs)
0    228.575273   228.550000
1    244.007632   238.945566
2    262.460323   255.620000
3    224.320352   224.220000
4    202.141848   199.710000

Difference

In [18]:
weights["DF"] = weights["Before (lbs)"] - weights["After (lbs)"]
weights.head()

Before (lbs)  After (lbs)        DF
0    228.575273   228.550000  0.025273
1    244.007632   238.945566  5.062066
2    262.460323   255.620000  6.840323
3    224.320352   224.220000  0.100352
4    202.141848   199.710000  2.431848

Stating the null hypothesis H0: 

1. H0: the mean DF is >= 0 
1. H1: the mean DF is < 0

Calculating the appropiate statistic

**Mean**

In [19]:
mean_w = round(weights["DF"].mean(), 2)
mean_w

2.51

**STD**

In [20]:
STD = round(weights["DF"].std(), 2)
STD

3.95

**SE**

In [21]:
SE = round(STD / np.sqrt(len(weights)), 2)
SE

1.25

In [22]:
T_stat = round((mean_w - 0) / SE, 2)
T_stat

2.01

Deciding if it is a one sided or two sided test: definetely it is a one sided because we want to know if we are going to lose weight with this program, > 0 otherwise it is the alternative H1

The P value is **0.038**

1. 0.038 < 0.10 Reject at 10% level of confidence we reject the H0, the program is not working
2. 0.038 < 0.05 Reject at 5% level of confidence we reject the H0, the program is not working
3. 0.038 > 0.01 Accept at 1% level of confidence we accept the H0, the program is working

Note that the t-score and the p-value are not affected by the unit of measurement wether it is kg or lbs. This shows the power of standardization.	

# S11_L65

Background: Ajax releases a new dish detergent, called Extreme Ajax+, which they advertise to be able to clean many more dishes with a single bottle.									

You want to see if that is true and have bought 25 bottles of Extreme Ajax+ and 25 bottles of Ajax and collected data on how many dirty dishes you can wash with a single bottle. (Population variance is assumed to be known.)									
										
1. Task 1	Find the standard deviations of the two samples								
2. Task 2	State the Null Hypothesis								
3. Task 3	Determine the Z-score								
4. Task 4	Find the p-value								
5. Task 5	Interpret the result								

In [23]:
ajax = pd.DataFrame({"Ajax" : [25, 115, 625], "Extreme Ajax" : [25, 100, 400]}, index = ["size", "mean", "variace"])
ajax["DF"] = ajax["Ajax"] - ajax["Extreme Ajax"]
ajax

Ajax  Extreme Ajax   DF
size       25            25    0
mean      115           100   15
variace   625           400  225

The difference of the variances of two population can be measured with a subtraction, let's change that

In [24]:
round(np.sqrt((625 / 25) + (400 / 25)), 2)

6.4

In [25]:
ajax.iloc[2, 2] = round(np.sqrt((625 / 25) + (400 / 25)), 2)

In [26]:
ajax

Ajax  Extreme Ajax    DF
size       25            25   0.0
mean      115           100  15.0
variace   625           400   6.4

**STD of the two samples**

In [27]:
np.sqrt(625)

25.0

In [28]:
np.sqrt(400)

20.0

**H0: ajax mean dishes cleaned - extreme ajax+ dishes cleaned > 0**

**Z score**

In [29]:
Z_score = round((115 - 100) / np.sqrt((625 / 25) + (400 / 25)), 2)
Z_score

2.34

**The P value in this case is 0.010**

1. 0.010 < 0.10 reject H0 
2. 0.010 < 0.05 reject H0 
3. 0.010 == 0.01 accept H0 at 1% confidence level we accept that extreme ajax+ clean more dishes per bottle

# S11_L67

You have data on the amount of times people click on a pop-up add on 24 Mondays and 21 Saturdays on an e-learning platform for several years. 

The samples are drawn independently.																			
																				
Task Statistically speaking, is there strong evidence that the number of clicks the add records on Mondays is higher than the number of clicks on Saturdays? 												

In [32]:
click = pd.DataFrame({"Monday" : [1078, 633, 24], "Saturday" : [908.20, 469.80, 21]}, index = ["Mean", "STD", "Sample_Size"])
click

Monday  Saturday
Mean           1078     908.2
STD             633     469.8
Sample_Size      24      21.0

1. H0: mean Monday - mean Saturday =< 0
2. H1: mean Monday - mean Saturday > 0

**Pooled variance formula**

In [36]:
variance_m = 633 ** 2
variance_s = 469.8 ** 2

In [37]:
variance_m

400689

In [38]:
variance_s

220712.04

In [41]:
pooled = round(((24 - 1) * (variance_m) + (21 - 1) * (variance_s)) / (24 + 21 - 2), 2)
pooled

316978.79

**We are going to use the T score** small sample sizes and population variance unknown

**Standard error**

In [44]:
SE = round(np.sqrt(pooled / 24 + pooled / 21), 2)
SE

168.23

In [47]:
T_score = round((1078 - 908.2) / SE, 2)
T_score

1.01

The P value is **0.16**

The p-value shows the reuslt is not significant. 								
Therefore, there is no reason to assume more people click on Mondays than on Saturdays.								